session.go

```diff
@@ -539,6 +537,7 @@ func (s *Session) mux(remoteBegin *frames.PerformBegin) {
 				// are safe to clean up its state.
 				delete(links, link.remoteHandle)
 				delete(deliveryIDByHandle, link.handle)
+				s.deallocateHandle(link)
```

Comment:
Just checking - was this moved here just because it made more sense to think of the session deleting a handle instead of doing it in the Sender/Receive mux?

Reply:
If the context expires/is cancelled when creating a Sender/Receiver, there's no mux running so we have to do it here.